bluechi: rename source code and build systems

dougsland opened the pull request:
hirte project will be renamed to bluechi.
This PR starts this process.

Fixes: https://github.com/containers/bluechi/issues/447
Fixes: https://github.com/containers/bluechi/issues/448
Fixes: https://github.com/containers/bluechi/issues/449
Fixes: https://github.com/containers/bluechi/issues/450
Fixes: https://github.com/containers/bluechi/issues/451
Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

dougsland committed:
bluechi: rename source code and build systems

hirte project is renamed to bluechi.

Fixes: https://github.com/containers/bluechi/issues/447
Fixes: https://github.com/containers/bluechi/issues/448
Fixes: https://github.com/containers/bluechi/issues/449
Fixes: https://github.com/containers/bluechi/issues/450
Fixes: https://github.com/containers/bluechi/issues/451
Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

rhatdan commented on LICENSES/README.md:
```diff
@@ -1,8 +1,8 @@
-# Hirte Project Licensing
+# Bluechi Project Licensing
```
C is capatalized I believe?

dougsland replied:
@rhatdan Sure, I will take care of all, regarding docs and md files. Thanks!

rhatdan commented on LICENSES/README.md:
```diff
@@ -1,8 +1,8 @@
 ## Main License
 
-The Hirte project uses single-line references to Unique License Identifiers as
+The Bluechi project uses single-line references to Unique License Identifiers as
```
BlueChi

rhatdan commented on LICENSES/README.md:
```diff
@@ -16,13 +16,13 @@ Unless otherwise noted, the systemd project sources are licensed under the terms
 
 New sources that cannot be distributed under **GPL-2.0-or-later** will no longer
-be accepted for inclusion in the Hirte project to maintain license uniformity.
+be accepted for inclusion in the Bluechi project to maintain license uniformity.
```
BlueChi

rhatdan commented on README.developer.md:
```diff
@@ -1,5 +1,5 @@
 
-# Hirte
+# Bluechi
```
BlueChi

rhatdan commented on README.developer.md:
```diff
@@ -12,9 +12,9 @@
-    - [hirtectl](#hirtectl)
+    - [bluechi](#bluechi)
+    - [bluechi-agent](#bluechi-agent)
+    - [bluechi-ctl](#bluechi-ctl)
```
Not sure we agreed on the '-' bluechictl vs bluechi-ctl.

dougsland replied:
The @mwperina doc says bluechi-ctl just following the doc.

engelmi replied:
Yes, the doc states `bluechi-ctl`, but it should be sth that had been discussed in the sync - not sure if this has been though. I'd prefer `bluechictl` and use the naming pattern of `kubectl` and `systemctl`.

mwperina replied:
I thought about the naming and `bluechi-ctl` seemed to easier to spell and more aligned with `bluechi-agent` or `bluechi-proxy` than `bluechictl`. As mentioned it's like that in the document and nobody complained so far.

mwperina replied:
OK, changed to `bluechictl`

rhatdan commented on README.developer.md:
```diff
@@ -135,12 +135,12 @@ In some cases, developers might need a debug session with tools like gdb, here a
 First, make sure **meson.build** contains **debug=true**.  
 
-Rebuild the hirte project with debug symbols included:
+Rebuild the bluechi project with debug symbols included:
```
BlueChi

rhatdan commented on README.developer.md:
```diff
@@ -244,7 +244,7 @@ After executing a `meson install` the MAN pages are located in the `man/man*` di
 ## Packaging
 
-Hirte is packaged as an RPM. The scripts used to create the SRPM and RPM may be found in [.build-scripts](./build-scripts).
+Bluechi is packaged as an RPM. The scripts used to create the SRPM and RPM may be found in [.build-scripts](./build-scripts).
```
BlueChi

rhatdan commented on README.maintainer.md:
```diff
@@ -1,8 +1,8 @@
-# Hirte
+# Bluechi
```
BlueChi

rhatdan commented on README.maintainer.md:
```diff
@@ -1,8 +1,8 @@
 ## Creating a new release
 
-Following actions need to be taken to release a new stable version of Hirte:
+Following actions need to be taken to release a new stable version of Bluechi:
```
BlueChi

rhatdan commented on README.md:
```diff
@@ -1,18 +1,18 @@
-# Hirte
+# Bluechi
```
BlueChi

rhatdan commented on README.md:
```diff
@@ -1,18 +1,18 @@
 [![Copr build status](https://copr.fedorainfracloud.org/coprs/mperina/hirte-snapshot/package/hirte/status_image/last_build.png)](https://copr.fedorainfracloud.org/coprs/mperina/hirte-snapshot/package/hirte/)
 
-Hirte is a systemd service controller intended for multi-node environments with
+Bluechi is a systemd service controller intended for multi-node environments with
```
BlueChi

rhatdan commented on README.md:
```diff
@@ -1,18 +1,18 @@
 traditional orchestration tools are not compliant with regulatory requirements.
 
-Hirte is relying on [systemd](https://github.com/systemd/systemd) and its D-Bus
+Bluechi is relying on [systemd](https://github.com/systemd/systemd) and its D-Bus
```
BlueChi

rhatdan commented on README.md:
```diff
@@ -1,18 +1,18 @@
 API, which it extends for multi-node use case.
 
-Hirte can also be used to control systemd services for containerized applications
+Bluechi can also be used to control systemd services for containerized applications
```
BlueChi

rhatdan commented on README.md:
```diff
@@ -21,33 +21,33 @@ to generate systemd service configuration to run a container via
-RPM packages for the Hirte project are available on
-[hirte-snapshot](https://copr.fedorainfracloud.org/coprs/mperina/hirte-snapshot/)
-COPR repo. To install hirte packages on your system please add that repo using:
+RPM packages for the Bluechi project are available on
```
BlueChi

rhatdan commented on README.md:
````diff
@@ -21,33 +21,33 @@ to generate systemd service configuration to run a container via
 ```
 
-When done you can install relevant hirte packages using:
+When done you can install relevant bluechi packages using:
````
BlueChi

rhatdan commented on bluechi.spec.in:
```diff
@@ -0,0 +1,227 @@
+
+Requires: systemd
+Recommends: bluechi-selinux
+
```
Needs to obsolete hirte package.

rhatdan commented on bluechi.spec.in:
```diff
@@ -0,0 +1,227 @@
+Recommends: bluechi-selinux
+
+%description
+Bluechi is a systemd service controller for multi-nodes environements with a
```
BlueChi

rhatdan commented on doc/man/bluechi-agent.1.md:
```diff
@@ -0,0 +1,89 @@
+
+## DESCRIPTION
+
+Bluechi is a systemd service controller intended for multi-node environment with a predefined number of nodes and with a focus on highly regulated environment such as those requiring functional safety (for example in cars).
```
BlueChi

rhatdan commented on selinux/bluechi.te:
```diff
@@ -0,0 +1,91 @@
+type bluechi_exec_t;
+init_daemon_domain(bluechi_t, bluechi_exec_t)
+
+type bluechi_agent_t;
```
type bluechi_agent_t alias hirte_agent_t;

mwperina commented on README.developer.md:
```diff
@@ -161,7 +161,7 @@ will produce a coverage report in `builddir/meson-logs/coveragereport/index.html
 All files related to the integration tests are located in [tests](./tests/) and are organized via
 [tmt](https://tmt.readthedocs.io/en/stable/). How to get started with running and developing
-integration tests for hirte is described in [tests/README](./tests/README.md).
+integration tests for bluechi is described in [tests/README](./tests/README.md).
```
BlueChi

mwperina commented on README.developer.md:
```diff
@@ -181,55 +181,55 @@ meson install -C builddir --destdir bin
-cp <builddir>/share/dbus-1/system.d/org.containers.hirte.Agent.conf /etc/dbus-1/system.d/
-cp <builddir>/share/dbus-1/system.d/org.containers.hirte.conf /etc/dbus-1/system.d/
+cp <builddir>/share/dbus-1/system.d/org.containers.bluechi.Agent.conf /etc/dbus-1/system.d/
+cp <builddir>/share/dbus-1/system.d/org.containers.bluechi.conf /etc/dbus-1/system.d/
```
`org.containers.hirte.` should be replaced with `io.github.eclipse-bluechi.bluechi` according to the doc and today's discussion

mwperina commented on README.developer.md:
```diff
@@ -181,55 +181,55 @@ meson install -C builddir --destdir bin
 Files for documentation of this project are located in the [doc](./doc/) directory comprising:
 
-- [api examples](./doc/api-examples/): directory containing python files that use the D-Bus API of hirte, e.g. for starting
+- [api examples](./doc/api-examples/): directory containing python files that use the D-Bus API of bluechi, e.g. for starting
```
BlueChi

mwperina commented on README.developer.md:
```diff
@@ -181,55 +181,55 @@ meson install -C builddir --destdir bin
 (see [Building MAN pages](#building-man-pages) for more information)
-- readthedocs files for building the documentation website of hirte (see [the README](./doc/README.md) for further information)
-- [diagrams.drawio](./doc/diagrams.drawio) file containing all diagrams used for hirte
+- readthedocs files for building the documentation website of bluechi (see [the README](./doc/README.md) for further information)
```
BlueChi

mwperina commented on README.developer.md:
```diff
@@ -181,55 +181,55 @@ meson install -C builddir --destdir bin
-- readthedocs files for building the documentation website of hirte (see [the README](./doc/README.md) for further information)
-- [diagrams.drawio](./doc/diagrams.drawio) file containing all diagrams used for hirte
+- readthedocs files for building the documentation website of bluechi (see [the README](./doc/README.md) for further information)
+- [diagrams.drawio](./doc/diagrams.drawio) file containing all diagrams used for bluechi
```
BlueChi

mwperina commented on README.md:
```diff
@@ -21,33 +21,33 @@ to generate systemd service configuration to run a container via
-[hirte-snapshot](https://copr.fedorainfracloud.org/coprs/mperina/hirte-snapshot/)
-COPR repo. To install hirte packages on your system please add that repo using:
+RPM packages for the BlueChi project are available on
+[bluechi-snapshot](https://copr.fedorainfracloud.org/coprs/mperina/hirte-snapshot/)
```
I'd leave it as `hirte-snapshot` until we resolve if to rename or switch to packit

mwperina commented on README.md:
```diff
@@ -21,33 +21,33 @@ to generate systemd service configuration to run a container via
+RPM packages for the BlueChi project are available on
+[bluechi-snapshot](https://copr.fedorainfracloud.org/coprs/mperina/hirte-snapshot/)
+COPR repo. To install bluechi packages on your system please add that repo using:
 
```
BlueChi

mwperina commented on README.md:
````diff
@@ -21,33 +21,33 @@ to generate systemd service configuration to run a container via
 
 ```bash
-dnf copr enable mperina/hirte-snapshot
+dnf copr enable mperina/bluechi-snapshot
````
This needs to remain `hirte-snapshot` until we resolve if to rename COPR repo or switch to packit

mwperina commented on bluechi.spec.in:
```diff
@@ -0,0 +1,230 @@
+%{_datadir}/dbus-1/interfaces/org.containers.bluechi.Job.xml
+%{_datadir}/dbus-1/interfaces/org.containers.bluechi.Manager.xml
+%{_datadir}/dbus-1/interfaces/org.containers.bluechi.Monitor.xml
+%{_datadir}/dbus-1/interfaces/org.containers.bluechi.Node.xml
```
`org.containers.hirte.` should be replaced with `io.github.eclipse-bluechi.bluechi` according to the doc and today's discussion

mwperina commented on bluechi.spec.in:
```diff
@@ -0,0 +1,230 @@
+%license LICENSE
+%{_bindir}/bluechi-agent
+%{_bindir}/bluechi-proxy
+%{_datadir}/dbus-1/system.d/org.containers.bluechi.Agent.conf
```
`org.containers.hirte.` should be replaced with `io.github.eclipse-bluechi.bluechi` according to the doc and today's discussion

mwperina commented on bluechi.spec.in:
```diff
@@ -0,0 +1,230 @@
+%{_bindir}/bluechi-proxy
+%{_datadir}/dbus-1/system.d/org.containers.bluechi.Agent.conf
+%{_datadir}/bluechi-agent/config/agent.conf
+%{_datadir}/dbus-1/interfaces/org.containers.bluechi.Agent.xml
```
`org.containers.hirte.` should be replaced with `io.github.eclipse-bluechi.bluechi` according to the doc and today's discussion

mwperina commented on bluechi.spec.in:
```diff
@@ -0,0 +1,230 @@
+%meson_test
+
+
+%changelog
```
Please leave changelog unchanged, hirte should remain there as is

mwperina commented on build-scripts/spdx-header-chech.sh:
```diff
@@ -8,13 +8,13 @@ CHECKED_FILES=".*\(\.c\|\.h\|\.sh\|\.xml\|meson\.build\)"
+# Used in src/libbluechi/common/list.h
 APPROVED_LICENSES="${APPROVED_LICENSES} LGPL-2.1-or-later"
-# Used in src/libhirte/ini/
+# Used in src/libbluechi/ini/
```
Ha, we forgot about it, now it's located in `subprojects/ini`

mwperina commented on build-scripts/spdx-header-chech.sh:
```diff
@@ -8,13 +8,13 @@ CHECKED_FILES=".*\(\.c\|\.h\|\.sh\|\.xml\|meson\.build\)"
+# Used in src/libbluechi/ini/
 APPROVED_LICENSES="${APPROVED_LICENSES} BSD-3-Clause"
-# Used in src/libhirte/hasmap/
+# Used in src/libbluechi/hasmap/
```
Ha, we forgot about it, now it's located in `subprojects/hashmap`

mwperina commented on data/meson.build:
```diff
@@ -3,11 +3,11 @@
+        'org.containers.bluechi.Manager.xml',
+        'org.containers.bluechi.Monitor.xml',
+        'org.containers.bluechi.Node.xml',
+        'org.containers.bluechi.Agent.xml'
```
org.containers.hirte. should be replaced with io.github.eclipse-bluechi.bluechi according to the doc and today's discussion

mwperina commented on data/meson.build:
```diff
@@ -23,14 +23,14 @@ conf = configuration_data()
-    output: 'org.containers.hirte.conf',
-    input: 'org.containers.hirte.conf.in',
+    output: 'org.containers.bluechi.conf',
+    input: 'org.containers.bluechi.conf.in',
```
org.containers.hirte. should be replaced with io.github.eclipse-bluechi.bluechi according to the doc and today's discussion

mwperina commented on data/meson.build:
```diff
@@ -23,14 +23,14 @@ conf = configuration_data()
-    output: 'org.containers.hirte.Agent.conf',
-    input: 'org.containers.hirte.Agent.conf.in',
+    output: 'org.containers.bluechi.Agent.conf',
+    input: 'org.containers.bluechi.Agent.conf.in',
```
org.containers.hirte. should be replaced with io.github.eclipse-bluechi.bluechi according to the doc and today's discussion

mwperina commented on data/org.containers.bluechi.Agent.conf.in:
```diff
@@ -3,22 +3,22 @@
 <busconfig>
 
   <policy user="@dbus_srv_user@">
```
Valid for the whole file: `org.containers.hirte.` should be replaced with `io.github.eclipse-bluechi.bluechi.` according to the doc and today's discussion

mwperina commented on data/org.containers.bluechi.Agent.xml:
```diff
@@ -1,7 +1,7 @@
 <!-- SPDX-License-Identifier: GPL-2.0-or-later -->
 <node>
-  <interface name="org.containers.hirte.Agent">
+  <interface name="org.containers.bluechi.Agent">
```
Valid for the whole file: `org.containers.hirte.` should be replaced with `io.github.eclipse-bluechi.bluechi.` according to the doc and today's discussion

mwperina commented on data/org.containers.bluechi.Manager.xml:
```diff
@@ -1,7 +1,7 @@
 <!-- SPDX-License-Identifier: GPL-2.0-or-later -->
 <node>
-  <interface name="org.containers.hirte.Manager">
+  <interface name="org.containers.bluechi.Manager">
```
Valid for the whole file: `org.containers.hirte.` should be replaced with `io.github.eclipse-bluechi.bluechi.` according to the doc and today's discussion

mwperina commented on data/org.containers.bluechi.Monitor.xml:
```diff
@@ -1,7 +1,7 @@
 <!-- SPDX-License-Identifier: GPL-2.0-or-later -->
 <node>
-  <interface name="org.containers.hirte.Monitor">
+  <interface name="org.containers.bluechi.Monitor">
```
Valid for the whole file: `org.containers.hirte.` should be replaced with `io.github.eclipse-bluechi.bluechi.` according to the doc and today's discussion

mwperina commented on data/org.containers.bluechi.Node.xml:
```diff
@@ -1,7 +1,7 @@
 <!-- SPDX-License-Identifier: GPL-2.0-or-later -->
 <node>
-  <interface name="org.containers.hirte.Node">
+  <interface name="org.containers.bluechi.Node">
```
Valid for the whole file: `org.containers.hirte.` should be replaced with `io.github.eclipse-bluechi.bluechi.` according to the doc and today's discussion

mwperina commented on data/org.containers.bluechi.conf.in:
```diff
@@ -3,22 +3,22 @@
 <busconfig>
 
   <policy user="@dbus_srv_user@">
```
Valid for the whole file: `org.containers.hirte.` should be replaced with `io.github.eclipse-bluechi.bluechi.` according to the doc and today's discussion

mwperina commented on data/org.containers.bluechi.internal.Agent.Metrics.xml:
```diff
@@ -1,7 +1,7 @@
 <!-- SPDX-License-Identifier: GPL-2.0-or-later -->
 <node>
-  <interface name="org.containers.hirte.internal.Agent.Metrics">
+  <interface name="org.containers.bluechi.internal.Agent.Metrics">
```
Valid for the whole file: `org.containers.hirte.` should be replaced with `io.github.eclipse-bluechi.bluechi.` according to the doc and today's discussion

mwperina commented on data/org.containers.bluechi.internal.Agent.xml:
```diff
@@ -1,7 +1,7 @@
 <!-- SPDX-License-Identifier: GPL-2.0-or-later -->
 <node>
-  <interface name="org.containers.hirte.internal.Agent">
+  <interface name="org.containers.bluechi.internal.Agent">
```
Valid for the whole file: `org.containers.hirte.` should be replaced with `io.github.eclipse-bluechi.bluechi.` according to the doc and today's discussion

mwperina commented on data/org.containers.bluechi.internal.Manager.xml:
```diff
@@ -1,9 +1,9 @@
 <!-- SPDX-License-Identifier: GPL-2.0-or-later -->
 <node>
-  <interface name="org.containers.hirte.internal.Manager">
+  <interface name="org.containers.bluechi.internal.Manager">
```
Valid for the whole file: `org.containers.hirte.` should be replaced with `io.github.eclipse-bluechi.bluechi.` according to the doc and today's discussion

mwperina commented on data/org.containers.bluechi.internal.Proxy.xml:
```diff
@@ -1,7 +1,7 @@
 <!-- SPDX-License-Identifier: GPL-2.0-or-later -->
 <node>
-  <interface name="org.containers.hirte.internal.Proxy">
+  <interface name="org.containers.bluechi.internal.Proxy">
```
Valid for the whole file: `org.containers.hirte.` should be replaced with `io.github.eclipse-bluechi.bluechi.` according to the doc and today's discussion

mwperina committed:
Fix renaming inconsistencies

Fixes: https://github.com/containers/bluechi/issues/447
Fixes: https://github.com/containers/bluechi/issues/448
Fixes: https://github.com/containers/bluechi/issues/449
Fixes: https://github.com/containers/bluechi/issues/450
Fixes: https://github.com/containers/bluechi/issues/451
Signed-off-by: Martin Perina <mperina@redhat.com>

mwperina committed:
Use org.eclipse.bluechi as a prefix for DBus APIs

Use org.eclipse.bluechi instead of io.github.eclipse-bluechi.bluechi as
a prefix for DBus APIs to satisfy DBus interface naming policy.

Fixes: https://github.com/containers/bluechi/issues/447
Fixes: https://github.com/containers/bluechi/issues/448
Fixes: https://github.com/containers/bluechi/issues/449
Fixes: https://github.com/containers/bluechi/issues/450
Fixes: https://github.com/containers/bluechi/issues/451
Signed-off-by: Martin Perina <mperina@redhat.com>

ygalblum commented on build-scripts/spdx-header-chech.sh:
```diff
@@ -8,15 +8,10 @@ CHECKED_FILES=".*\(\.c\|\.h\|\.sh\|\.xml\|meson\.build\)"
-# Used in src/libhirte/common/list.h
+# Used in src/libbluechi/common/list.h
 APPROVED_LICENSES="${APPROVED_LICENSES} LGPL-2.1-or-later"
-# Used in src/libhirte/ini/
```
Why were these removed?

mwperina replied:
When we had those files directly in the repo, I have added an SPDX header to show their license. But when we moved to use those as subprojects, it's no longer needed, because we don't check SPDX header for subprojects: 

https://github.com/containers/bluechi/blob/main/build-scripts/spdx-header-chech.sh#L25

alexlarsson commented on bluechi.spec.in:
```diff
@@ -110,61 +112,61 @@ Requires(postun): policycoreutils-python-utils
-%package -n python3-pyhirte
-Summary: Python bindings for Hirte
+%package -n python3-bluechi
+Summary: Python bindings for BlueChi
```
This changes from `pyhirte`to `bluechi`. not `pybluechi`. I think this is fine, just checking to see that this was the intention?

mwperina replied:
Yes, most of packages in PIP today are named without `py` prefix, so fixing the name as a part of project rename is good time (it needs to be said that PIP doesn't support renaming packages, so bluechi will be a new package in PIP)

mwperina replied:
I've also mentioned that in the rename google doc file

alexlarsson commented on src/libbluechi/common/cfg.c:
```diff
@@ -439,12 +439,12 @@ int cfg_agent_def_conf(struct config *config) {
         int result = 0;
 
-        result = cfg_set_default_section(config, CFG_SECT_HIRTE);
+        result = cfg_set_default_section(config, CFG_SECT_BC);
```
I would perhaps use the full bluechi name for this one.

mwperina replied:
Fixed

alexlarsson reviewed (approved):
This looks good to me, any further fixes can land later.
